src/managed/Microsoft.DotNet.Build.Bundle/Bundler.cs

```diff
@@ -0,0 +1,205 @@
+        /// </summary>
+        /// <returns>Returns the offset of the start 'file' within 'bundle'</returns>
+
+        long AddToBundle(Stream bundle, Stream file, FileType type = FileType.Extract)
+        {
+            // Allign assemblies, since they are loaded directly from bundle
+            if (type == FileType.Assembly)
+            {
+                long padding = AssemblyAlignment - (bundle.Position % AssemblyAlignment);
+                bundle.Position += padding;
+            }
+
+            file.Position = 0;
+            long startOffset = bundle.Position;
+            file.CopyTo(bundle);
+
+            return startOffset;
+        }
+
+        bool ShouldEmbed(string fileName)
+        {
+            if (fileName.Equals(HostName))
+            {
+                // The bundle starts with the host, so ignore it while embedding.
+                return false;
+            }
+
+            if (fileName.Equals(RuntimeConfigDevJson))
+            {
+                // Ignore the machine specific configuration file.
+                return false;
+            }
+
+            if (Path.GetExtension(fileName).ToLower().Equals(".pdb"))
+            {
+                return EmbedPDBs;
+            }
+
+            return true;
+        }
+
+        FileType InferType(string fileName, Stream file)
+        {
+            if (fileName.Equals(DepsJson))
+            {
+                return FileType.DepsJson;
+            }
+
+            if (fileName.Equals(RuntimeConfigJson))
+            {
+                return FileType.RuntimeConfigJson;
+            }
+
+            if (fileName.Equals(Application))
+            {
+                return FileType.Application;
+            }
+
+            try
+            {
+                PEReader peReader = new PEReader(file);
+                CorHeader corHeader = peReader.PEHeaders.CorHeader;
+                if ((corHeader != null) && ((corHeader.Flags & CorFlags.ILOnly) != 0))
+                {
+                    return FileType.Assembly;
+                }
+            }
+            catch (BadImageFormatException)
+            {
+            }
+
+            return FileType.Extract;
+        }
+
+        void GenerateBundle()
+        {
+            string bundlePath = Path.Combine(OutputDir, HostName);
+
+            if (File.Exists(bundlePath))
+            {
+                Program.Log($"Ovewriting existing File {bundlePath}");
+            }
+
+            // Start with a copy of the host executable.
+            // Copy the file to preserve its permissions.
+            File.Copy(Path.Combine(ContentDir, HostName), bundlePath, overwrite: true);
+
+            using (BinaryWriter writer = new BinaryWriter(File.OpenWrite(bundlePath)))
+            {
+                Stream bundle = writer.BaseStream;
+                Manifest manifest = new Manifest();
+
+                bundle.Position = bundle.Length;
+                foreach (string filePath in Directory.GetFiles(ContentDir))
```

Comment:
Should all input files be sorted so the output file is deterministic?

Reply:
What about files in sub directories? Shouldn't they also be bundled?

Reply:
Thanks for the review @0xd4d. 
The bundler is expected to be used with rid-specific publish operations, where all the published files will be in the same directory. So, I don't know if a use-case for sub-directories yet.

I'll fix the other issues you'be pointed out. Thanks.

Reply:
Shouldn't all app files should be bundled? Some of them could be in sub directories because not everyone wants to put everything in the base dir.